crates/matrix-sdk/src/widget/client/actions.rs

```diff
@@ -14,11 +14,11 @@
 
 use std::{borrow::Cow, error::Error, ops::Deref};
 
-use ruma::events::TimelineEventType;
+use ruma::events::{MessageLikeEventType, StateEventType, TimelineEventType};
 use serde::{Deserialize, Serialize};
 use serde_json::Value as JsonValue;
 
-use crate::widget::Permissions;
+use crate::widget::{Permissions, StateKeySelector};
 
 /// Action (a command) that client (driver) must perform.
```

Comment:
(predating your changes, but... what does this comment mean? xD It's quite confusing with the words in parenthesis that refer to other concepts)

Reply:
This is going to be rewritten...